src/createServer.js

```diff
@@ -1,8 +1,99 @@
 'use strict';
 
+const url = require('url');
+const http = require('http');
+const fs = require('fs');
+const path = require('path');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  return http.createServer((req, res) => {
+    const normalizeUrl = new url.URL(req.url, `http://${req.headers.host}`);
+    const pathName = normalizeUrl.pathname;
+    const arrayFromPathName = pathName.slice(1).split('/');
+
+    if (pathName.includes('//')) {
+      res.writeHead(404, {
+        'Content-Type': 'text/plain',
+      });
+
+      res.end(
+        JSON.stringify({
+          error: 'Path to file should be like this /file/folderName/filename',
+        }),
+      );
+
+      return;
+    }
+
+    if (!pathName.startsWith('/file')) {
+      res.writeHead(400, {
+        'Content-Type': 'text/plain',
+      });
+
+      res.end(
+        JSON.stringify({
+          error: 'Path to file should be like this /file/folderName/filename',
+        }),
+      );
+
+      return;
+    }
+
+    if (arrayFromPathName.includes('..')) {
+      res.writeHead(400, {
+        'Content-Type': 'text/plain',
+      });
+
+      res.end(
+        JSON.stringify({
+          error: 'Path to file should be like this /file/folderName/filename',
+        }),
+      );
+    }
+
+    if (arrayFromPathName.includes('')) {
+      res.writeHead(404, 'File not found', {
+        'Content-Type': 'text/plain',
+      });
+
+      res.end(
+        JSON.stringify({
+          error: 'Path to file should be like this /file/folderName/filename',
+        }),
+      );
+    }
+
+    const slicedPathName = arrayFromPathName.slice(1).join('/');
+
+    if (fs.existsSync(`./public/${slicedPathName}`)) {
+      res.writeHead(404, 'File not found', {
+        'Content-Type': 'text/plain',
+      });
+
+      res.end(
+        JSON.stringify({
+          error: 'Path does not exist',
+        }),
+      );
+    }
+
+    const filePath = path.join(__dirname, 'public', slicedPathName);
+
+    fs.readFile(filePath, 'utf-8', (error, data) => {
```

Comment:
folder public shouldn't be inside src
```suggestion
    fs.readFile(`./public/${slicedPathName}`, 'utf-8', (error, data) => {
```